### PR TITLE
Cache removed and filelist unit test

### PR DESCRIPTION
**Pull request by SeanTConrad**
## Summary by Sourcery

Add a unit test for file list processing and remove caching from the main function to streamline the code.

Enhancements:
- Remove the caching mechanism from the main function in modernmetric to simplify the code.
- Updated workflows so coverage runs. Codecov upload is failing, but that's because of the token
- New unit test found some bugs in Python 3.9 with "|" operator that I fixed

Tests:
- Add a new unit test 'test_filelist_scan' to verify the functionality of processing a JSON file list of file paths.

### CODE REVIEW

File: modernmetric/license.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 requestx = httpx.Client(http2=True, timeout=None)
 
 
-def report(identifier: str | int, product: str, die: bool = False):
+def report(identifier: "str | int", product: str, die: bool = False):
```
Comment (aylusltd) [inline modernmetric/license.py:6]: do this as a union imported from typing

File: modernmetric/__main__.py [hunk 2]
```diff
@@ -147,7 +147,7 @@ def ArgParser(custom_args=None):
 # e.g. ["--file=path/to/filelist.json"]
 
 
-def main(custom_args=None, license_identifier: str | int = None):
+def main(custom_args=None, license_identifier: "str | int" = None):
```
Comment (aylusltd) [inline modernmetric/__main__.py:150]: use a union

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -39,11 +39,5 @@ jobs:
           python -m pip install --upgrade pip
           pip install -r requirements.txt
 
-      - name: Run Pytest
+      - name: Run Pytest with coverage
```
Comment (aylusltd) [inline .github/workflows/test.yml:42]: aren't you running pytest twice

File: modernmetric/license.py (hunk 1 above)
Comment (SeanTConrad) [reply modernmetric/license.py]: fixed

File: .github/workflows/test.yml (hunk 3 above)
Comment (SeanTConrad) [reply .github/workflows/test.yml]: Not exactly. test.yml runs PyTest 8 times. Mac and Linxux x 4 Python versions. I moved Codecov out, so it doesn't upload the file 8 times, but it still needs to run pytest once to make the coverage file

**Review by aylusltd — APPROVED** [review]

**Pull request merged**